ADR: extend IstioCR authorizers with pathPrefix and timeout options (#1610)
kyma-project/istio: docs/contributor/adr/0011-extend-external-authorizer-path-timeout.md

# ADR 11: Extend External Authorizer definition in the IstioCR to include pathPrefix and timeout configurations

## Status
<!--- Specify the current state of the ADR, such as whether it is proposed, accepted, rejected, deprecated, superseded, etc. -->
Accepted

## Context
<!--- Describe the issue or problem that is motivating this decision or change. -->

> The current implementation of the External Authorizer in IstioCR lacks the ability to specify a custom path prefix and timeout for the authorization requests.
> This limitation can lead to challenges in integrating with external authorization services that require specific endpoints or have different performance characteristics.

Reviewer: ```suggestion
The current implementation of the External Authorizer in the Istio custom resource (CR) lacks the ability to specify a custom path prefix and timeout for the authorization requests.
This limitation can lead to challenges in integrating with external authorization services that require specific endpoints or have different performance characteristics.
```

## Decision
<!--- Explain the proposed change or action and the reason behind it. -->

> To enhance the flexibility and usability of the External Authorizer in IstioCR, we propose to extend its configuration options to include:

Reviewer: ```suggestion
To enhance the flexibility and usability of the External Authorizer in the Istio CR, we propose to extend its configuration options to include:
```

> 1. **pathPrefix**: Allow users to specify a custom path prefix for the authorization requests. For example if the external authorizer service expects requests at `/auth/users`, while requests to the end service are sent to `/users`, the pathPrefix can be set to `/auth` to ensure correct routing.

Reviewer: ```suggestion
1. **pathPrefix**: Allows users to specify a custom path prefix for the authorization requests. For example, if the external authorizer service expects requests at `/auth/users`, while requests to the end service are sent to `/users`, the pathPrefix can be set to `/auth` to ensure correct routing.
```

> 2. **timeout**: Introduce a timeout setting to define how long the system should wait for a response from the external authorizer before considering the request as failed.

Reviewer: ```suggestion
2. **timeout**: Introduces a timeout setting to define how long the system should wait for a response from the external authorizer before considering the request as failed.
```

> Above fields are exposed as one to one mapping of the Istio Mesh Config configuration. They are optional and if not set, the values will fall back to Istio's defaults.

Reviewer: ```suggestion
The above fields are exposed as a one-to-one mapping of the Istio Mesh Config configuration. They are optional, and if not set, the values fall back to Istio's defaults.
```

> These additions will enable better integration with a wider range of external authorization services and improve the overall reliability of the authorization process.

Reviewer: ```suggestion
These additions enable better integration with a wider range of external authorization services and improve the overall reliability of the authorization process.
```

> API structure after the change:

Reviewer: ```suggestion
See the API structure after the change:
```
